R/convenience-functions.R

```diff
@@ -0,0 +1,194 @@
+#' the forecasts and observations and then apply the score. Applying a
+#' transformation after the score risks losing propriety of the proper scoring
+#' rule.
+#'
+#' @inheritParams score
+#' @param fun A function used to transform both true values and predictions.
+#' If `fun = log` it automatically checks internally whether there are any
+#' zero values. If so, the function tries to apply `log(x + 1)` instead and
+#' gives a warning. You can pass an argument `offset = 1` (or any other value)
+#' to the function to avoid the warning.
+#' @param append whether or not to append a transformed version of the data to
+#' the currently existing data (default is TRUE). If selected, the data gets
+#' transformed and appended to the existing data frame, making it possible to
+#' use the outcome directly in `score()`. An additional column, 'scale', gets
+#' created that denotes which rows or untransformed ('scale' has the value
+#' "natural") and which have been transformed ('scale' has the value passed to
+#' the argument `label`).
+#' @param label A string for the newly created 'scale' column to denote the
+#' newly transformed values. Only relevant if `append = TRUE`.
+#' @param ... Additional parameters to pass to the function you supplied. If
+#' you're using `fun = log`, one parameter added by default is the option to
+#' set `offset = 1` (or any other number) to apply the function
+#' `log (1 + offset)` instead (useful if you have e.g zero values in the data).
+#' @return A data.table with either a transformed version of the data, or one
+#' with both the untransformed and the transformed data. includes the original data as well as a
+#' transformation of the original data. There will be one additional column,
+#' 'scale', present which will be set to "natural" for the untransformed
+#' forecasts.
+#'
+#' @importFrom data.table ':=' is.data.table copy
+#' @author Nikos Bosse \email{nikosbosse@@gmail.com}
+#' @export
+#' @references Transformation of forecasts for evaluating predictive
+#' performance in an epidemiological context
+#' Nikos I. Bosse, Sam Abbott, Anne Cori, Edwin van Leeuwen, Johannes Bracher,
+#' Sebastian Funk
+#' medRxiv 2023.01.23.23284722
+#' \doi{https://doi.org/10.1101/2023.01.23.23284722}
+#' <https://www.medrxiv.org/content/10.1101/2023.01.23.23284722v1> # nolint
+
+#' @keywords check-forecasts
+#' @examples
+#'
+#' # add log transformed forecasts (produces a warning as some values are zero)
+#' transform_forecasts(example_quantile, truncate = TRUE, append = FALSE)
+#'
+#' # specifying an offset manually for the log transformation removes the warning
+#' transform_forecasts(example_quantile, truncate = TRUE, offset = 1)
+#'
+#' # adding multiple transformations
+#' library(magrittr) # pipe operator
+#' example_quantile %>%
+#'   # truncate all negative values for both log and sqrt
+#'   transform_forecasts(fun = function(x) pmax(0, x), append = FALSE) %>%
```

Comment:
out of date?

Reply:
It's needed as the sqrt function doesn't have a truncate option. That part is maybe a bit confusing... The truncate option is part of `log_shift()`, not of `transform_forecasts()`

Reply:
I changed the example to this: 
```
#' example_quantile %>%
#'   transform_forecasts(offset = 1, truncate = TRUE) %>%
#'   # manually truncate all negative values before sqrt
#'   transform_forecasts(fun = function(x) pmax(0, x), append = FALSE) %>%
#'   transform_forecasts(fun = sqrt, label = "sqrt") %>%
#'   score() %>%
#'   summarise_scores(by = c("model", "scale"))
```

maybe that's a bit more intuitive (it also helped me catch an error where if you chose append = FALSE it would override previous transformations)